app/models/journaled/writer.rb

```diff
@@ -27,20 +27,22 @@ def initialize(journaled_event:)
   def journal!
     base_event_json_schema_validator.validate! serialized_event
     json_schema_validator.validate! serialized_event
-    Journaled.enqueue!(journaled_delivery, journaled_enqueue_opts)
+    Journaled::DeliveryJob
+      .set(journaled_enqueue_opts.reverse_merge(priority: Journaled.job_priority))
```

Comment:
This means that `enqueue_with` will need to consist of ActiveJob-compatible `set` options. So `queue` and `priority` should still work, but `run_at` will need to become `wait_until`, and other custom parameters may require extending the ActiveJob adapter in order to pass them through to the underlying queue backend.